code/src/components/products-container/ProductsContainer.tsx

```diff
@@ -36,8 +45,18 @@ const ProductsContainer = ({
 
   const skeletonCards = repeatComponent(<ProductSkeleton />, loadingItemsCount);
 
-  const handleAddToCart = () => {
-    openDrawer(<CartDrawer />);
+  const handleAddToCart = async (product: Product) => {
+    try {
+      if (user?.id) {
+        await addToCart({ productId: product.id, userId: user.id }).unwrap();
+        openDrawer(<CartDrawer />);
+      }
+    } catch (e) {
+      openSnackbar({
```

Comment:
Please consider using `openSnackbarWithTimeout` here if we need autohiding functionality. Same in other places

Reply:
Thanks, used suggested approach